### ISSUE TITLE
addl changes for AtlasVolunteer role

### ISSUE THREAD

**Issue by atlasjen** [Atlas-Assistance-Dogs/atlas-dogs#492]
let's also hide any contact info fields for anyone who is not a volunteer
is this doable? i know salesforce uses these to dedupe, so not sure if that is an issue.
this would apply to contact and account.

i am thinking we hide these on contact:
email, personal email, phone, other phone, alternate email , birthdate, pronoun, specified pronoun, mailing address, other address
and hide these on account:
phone, billing address, shipping address

**Comment by deb761** [comment]
The fields are hidden by the user has access.  If we want to hide fields based on position, we need to move them into an action, like we did for the position-specific data.

**Comment by atlasjen** [comment]
i'm not sure what moving to an action involves.
would they still be on the main page or would they be moved to a separate tab?
I should also clarify anyone who is in the Volunteer permission set you are working on.. not based on position.

**Comment by deb761** [comment]
So you want these fields to be visible in the permission set for contacts with certain positions, but not others?  That's the challenging part.

**Comment by deb761** [comment]
How about we use sharing rules to hide the non-volunteer contacts, and create a LWC that shows emergency contact information?  Or is that even necessary?  If we treat share the emergency contacts with the volunteers?

**Comment by atlasjen** [comment]
Using Dev branch in a scratch org:

I created new user w Atlas Minimum Access Profile. Note, i did not make them marketing users.
I manually added them to AtlasBase and AtlasVolunteer permissions sets

Verified good:
* i can see all contacts but cannot access Team, Background Check, or Dog objects.
* No demographic info is showing 
* notes and activities come through as expected
* Only volunteer position details show
* tested deleting Accounts and Contacts and it fails with permission error.
* tested that i couldn't edit programs
* tested that i could edit program assignments

Will need to test in prod
* we don't have our custom reports, so i will need to see which this user can or can't access. they can access the reports presently there

### **Needs changing:**

1.  We are missing related people, for example, their emergency contacts

2. I was able to delete a lead owned by someone else. I am ok w/ them deleting leads owned by them, but not by someone else.
3.  Volunteer files don't show (blank and NA category do). Note, files classified as Trainer properly don't show. i think keeping blank and NA is good, but we should also include Volunteer.

Here is what i uploaded for a user:
![image](https://github.com/Atlas-Assistance-Dogs/atlas-dogs/assets/66139437/1e95dcd9-13ae-44c6-a558-86c8af26826b)

Here is what i see as the volunteer user:
![image](https://github.com/Atlas-Assistance-Dogs/atlas-dogs/assets/66139437/53dc8114-ed9a-4e38-bed4-b90c362b28fa)

**Comment by deb761** [comment]
1. Mmm.  I'll look into that.  They should have been added to the Volunteer group.
2. That's a factor of the scratch org having Leads as public rather than private.  Try changing the setting in the scratch org.  The package doesn't affect that at all.
3. That sounds like either we're missing a controller permission, or the CategoryRules aren't set properly.

**Comment by atlasjen** [comment]
let me know when this is ready to retest :)

**Comment by deb761** [comment]
Sigh, I suspect including the groups in the package will be problematic.

**Comment by deb761** [comment]
It's ready to test, I have a PR for it.  But you will need to enable the relationship trigger.

**Comment by atlasjen** [comment]
Tested again.

- I created new user (tuser@atlasdog.org) w Atlas Minimum Access Profile. Note, i did not make them marketing users.

- I manually added them to AtlasBase and AtlasVolunteer permissions sets from the visual code terminal

> sfdx force:user:permset:assign -n AtlasVolunteer -o tuser@atlasdog.org
> sfdx force:user:permset:assign -n AtlasBase -o tuser@atlasdog.org     

- I went to Setting, Apex Triggers, and activated the Relatipnship Trigger
- I logged in as my test user:

- I didn't see any contacts at all.  [Note they are there for my default user and there are plenty w position of volunteer]
I also don't see Accounts. I can see Program Assignments, Programs, Leads.

- I created a test user "bob brown" bbrown@user.com

I got a popup error about a duplicate user but it made it anyway. I was able to create another contact who was a relationship and see them. So I think we can mark off relationships as working ok.  I went back and checked as my scratch user and no other email matched, so not sure what is going on there.

-  I tried to add files but still not seeing "Volunteer" as a category option for my active volunteer. 

![image](https://github.com/Atlas-Assistance-Dogs/atlas-dogs/assets/66139437/f9ab35a5-2d9d-4e05-9f30-b38e7346e3e9)

- I logged out as my test user and added two files to Bob Brown, one a volunteer type and one another.
- I logged back in as my test user. I still can't see any files that i added.


### So in summary, what needs examining

1. Why aren't seeing ANY contacts or accounts at all showing up for the atlas volunteer user
2. Same issue as item 3 above, Volunteer files don't show. I can't see ones already added and I can't add them.

**Comment by deb761** [comment]
1. Sharing Rules need to be defined.  I was unable to include them in the project because:

> Error: Could not process MDAPI response: Update of SharingCriteriaRule Contact.Volunteer: Error on line 3, col 27: Deploying sharing
> rule Volunteer not supported for  object Contact since it's org wide default is 'Controlled By Parent'

2. You will need to create the Sharing Rule to test this.

**Comment by atlasjen** [comment]
reopening in 1.8

**Comment by deb761** [comment]
Closing this because it is addressed in PR #536.  If there are more things to address, please open a new issue.  This re-opening of issues makes it hard for me to keep track of what I've worked on.  We can always reference closed issues when creating new if we need to refer to history.

Its frustrating when an issue is reopened because I forgot to setup something in a scratch org.

And yes, I might change my mind about this, too.
